name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/pdf/MerkurPrivatBankPDFExtractor.java

```diff
@@ -0,0 +1,115 @@
+package name.abuchen.portfolio.datatransfer.pdf;
+
+import static name.abuchen.portfolio.util.TextUtil.trim;
+
+import name.abuchen.portfolio.datatransfer.pdf.PDFParser.Block;
+import name.abuchen.portfolio.datatransfer.pdf.PDFParser.DocumentType;
+import name.abuchen.portfolio.datatransfer.pdf.PDFParser.Transaction;
+import name.abuchen.portfolio.model.BuySellEntry;
+import name.abuchen.portfolio.model.Client;
+import name.abuchen.portfolio.model.PortfolioTransaction;
+
+@SuppressWarnings("nls")
+public class MerkurPrivatBankPDFExtractor extends AbstractPDFExtractor
+{
+
+    public MerkurPrivatBankPDFExtractor(Client client)
+    {
+        super(client);
+        
+        // @formatter:off
+        // Am Marktplatz 10 · 97762 Hammelburg
+        // @formatter:on
+        addBankIdentifier("Am Marktplatz 10 · 97762 Hammelburg"); //$NON-NLS-1$
+
+        addBuySellTransaction();
+    }
+
+    @Override
+    public String getLabel()
+    {
+        return "MERKUR PRIVATBANK KGaA"; //$NON-NLS-1$
+    }
+
+    private void addBuySellTransaction()
+    {
+        DocumentType type = new DocumentType("Kauf");
+        this.addDocumentTyp(type);
+
+        Transaction<BuySellEntry> pdfTransaction = new Transaction<>();
+        pdfTransaction.subject(() -> {
+            BuySellEntry entry = new BuySellEntry();
+            entry.setType(PortfolioTransaction.Type.BUY);
+            return entry;
+        });
+        
+        // Start from: Wertpapier Abrechnung Kauf
+        Block firstRelevantLine = new Block("^Wertpapier Abrechnung Kauf.*$");
+        type.addBlock(firstRelevantLine);
+        firstRelevantLine.set(pdfTransaction);
+        
+        pdfTransaction
+                        // @formatter:off
+                        // Type (Optional)
+                        // Wertpapier Abrechnung Kauf
+                        // @formatter:on
+                        .section("type").optional().match("^Wertpapier Abrechnung (?<type>Kauf).*$").assign((t, v) -> {
+                            if (v.get("type").equals("Kauf"))
+                                t.setType(PortfolioTransaction.Type.BUY);
+                        })
+                        // @formatter:off
+                        // Security identification:
+                        // Stück 125,3258 XTR.(IE) - MSCI WORLD              IE00BJ0KDQ92 (A1XB5U)
+                        // REGISTERED SHARES 1C O.N.
+                        // @formatter:on
+                        .section("name", "ln2", "isin", "wkn")
```

Comment:
The `currency` attribute is missing here.

We create the securities in the traded extracted currency.
You can look at this --> [here](https://github.com/buchen/portfolio/blob/84a1a6d5cf400f4604e8982210ef5503997b4b4c/name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/pdf/AbstractPDFExtractor.java#L193:L197) <-- in the source code and also in the Contributing rules --> [here](https://github.com/buchen/portfolio/blob/master/CONTRIBUTING.md#naming-conventions-for-detected-values) <-- under Security identification.

Add `currency` to the section attributes,
Add this line
`.match("^Ausf.hrungskurs [\\.,\\d]+ (?<currency>[\\w]{3}) .*$")`